lib/shared/index.ts

```diff
@@ -108,10 +108,12 @@ export class Shared extends Construct {
       });
 
       // Create VPC Endpoint for SageMaker Runtime
-      vpc.addInterfaceEndpoint("SageMakerRuntimeEndpoint", {
-        service: ec2.InterfaceVpcEndpointAwsService.SAGEMAKER_RUNTIME,
-        open: true,
-      });
+      if (props.config.llms.sagemaker.length > 0) {
```

Comment:
```suggestion
      if (props.config.llms.sagemaker.length > 0 || props.config.rag.enabled) {
```
At this time, If rag is enabled, it will automatically create a sagemaker endpoint for embedding/cross encoding regardless of the config. (unless I am mistaken?)